File: Crouch.Core.Model/Order.cs
namespace Crouch.Core.Model
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Diagnostics.CodeAnalysis;

    public partial class Order
    {
        [SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Order()
        {
            OrderItems = new HashSet<OrderItems>();
        }

        [DisplayName("Order ID")]
        public int OrderId { get; set; }

        [DisplayName("Order Status")]
        public int OrderStatusId { get; set; }

        [DisplayName("Customer ID")]
        public int CustomerId { get; set; }

        [DisplayName("Order Title")]
        public string OrderTitle { get; set; }

        public DateTime OrderDate { get; set; }

        public DateTime CollectionDate { get; set; }

        [DisplayName("Deposit")]
        public decimal OrderDeposit { get; set; }

        [DisplayName("Order Value")]
        public decimal? OrderValue { get; set; }

        [DisplayName("Order Note")]
        public string OrderNote { get; set; }

        public virtual Customer Customer { get; set; }

        public virtual OrderStatus OrderStatus { get; set; }

        public virtual ICollection<OrderItems> OrderItems { get; set; }
    }
}

File: Crouch.UI/src/app/model/order.ts
import { Customer } from '../model/customer';
import { OrderItem } from '../model/order-item';

export class Order {
    public orderId: number;
    public orderStatusId: number;
    public customerId: number;
    public orderTitle: string;
    public orderDate: Date;
    public collectionDate: Date;
    public orderDeposit: number;
    public orderValue: number;
    public customer: Customer;
    public orderItems: OrderItem[];
}
